libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -794,6 +797,14 @@ class MapboxRouteLineApi(
         routeProgress: RouteProgress,
         consumer: MapboxNavigationConsumer<Expected<RouteLineError, RouteLineUpdateValue>>
     ) {
+        if (settingNewRoutes) {
```

Comment:
Why not wrap this in the `mutex` instead of introducing the flag?

Reply:
`updateWithRouteProgress` and `updateTravelledRouteLine` are not wrapped in mutex, I'm not sure why, but I assumed it had something to do with them being invoked quite often. That we wanted to avoid delays in that case or sth like that. Do you have more context on that?